src/unfold.rs

```diff
@@ -113,19 +106,20 @@ impl<T, F, Fut, It> Stream for Unfold<T, F, Fut>
     type Item = It;
 
     fn poll_next(
-        mut self: Pin<&mut Self>,
+        self: Pin<&mut Self>,
         cx: &mut Context<'_>,
     ) -> Poll<Option<It>> {
-        if let Some(state) = self.as_mut().state().take() {
-            let fut = (self.as_mut().f())(state);
-            self.as_mut().fut().set(Some(fut));
+        let this = self.project();
```

Comment:
`self.project` is introduced by `pin_project` and allows certain fields to be mutable, while others are kept pinned. This replaces the use of `pin_utils` and removes calls to `unsafe`.